packages/snap-preact-components/src/components/Organisms/HorizontalFacets/HorizontalFacets.tsx

```diff
@@ -0,0 +1,283 @@
+			// inherited props
+			...defined({
+				disableStyles,
+				overlay,
+			}),
+			// component theme overrides
+			theme: props?.theme,
+		},
+		MobileSidebar: {
+			// default props
+			className: 'ss__horizontal-facets__header__mobile-sidebar',
+			hidePerPage: true,
+			hideSortBy: true,
+			// global theme
+			...globalTheme?.components?.mobileSidebar,
+			// inherited props
+			...defined({
+				disableStyles,
+			}),
+			// component theme overrides
+			theme: props?.theme,
+		},
+	};
+
+	const styling: { css?: StylingCSS } = {};
+	const stylingProps = props;
+
+	if (styleScript && !disableStyles) {
+		styling.css = [styleScript(stylingProps), style];
+	} else if (!disableStyles) {
+		styling.css = [CSS.facets(stylingProps), style];
+	} else if (style) {
+		styling.css = [style];
+	}
+
+	const [selectedFacet, setSelectedFacet] = useState<IndividualFacetType | undefined>(undefined);
+
+	const innerRef = useClickOutside(() => {
+		selectedFacet && setSelectedFacet(undefined);
+	});
+
+	return facetsToShow && facetsToShow?.length > 0 ? (
+		<CacheProvider>
+			<div
+				className={classnames('ss__horizontal-facets', { 'ss__horizontal-facets--overlay': overlay }, className)}
+				ref={innerRef as React.LegacyRef<HTMLDivElement>}
+				{...styling}
+			>
+				<div className="ss__horizontal-facets__header">
+					{facetsToShow.map((facet: IndividualFacetType) => (
+						<Dropdown
+							{...subProps.dropdown}
+							open={selectedFacet?.field === facet.field}
+							onClick={() => {
+								if (selectedFacet === facet) {
+									setSelectedFacet(undefined);
+									return;
+								}
+								setSelectedFacet(facet);
+							}}
+							button={
+								<div
+									className="ss__horizontal-facets__header__dropdown__button ss__dropdown__button"
+									role="heading"
+									aria-level={3}
+									aria-label={`currently ${selectedFacet?.field === facet.field ? 'collapsed' : 'open'} ${facet.field} facet dropdown ${
+										(facet as ValueFacet).values?.length ? (facet as ValueFacet).values?.length + ' options' : ''
+									}`}
+								>
+									{facet?.label}
+									<Icon {...subProps.icon} icon={selectedFacet?.field === facet.field ? 'angle-up' : 'angle-down'} />
+								</div>
+							}
+							disableOverlay={!overlay}
+						>
+							{overlay ? <Facet {...subProps.facet} facet={facet} /> : undefined}
+						</Dropdown>
+					))}
+					{(isOverflowing || alwaysShowFiltersButton) && <MobileSidebar controller={controller as any} {...subProps.MobileSidebar}></MobileSidebar>}
+				</div>
+
+				{!overlay && selectedFacet && (
+					<div
+						className={classnames(
+							'ss__horizontal-facets__content',
+							`ss__horizontal-facets__content--${selectedFacet.display}`,
+							`ss__horizontal-facets__content--${selectedFacet.field}`
+						)}
+					>
+						<Facet {...subProps.facet} name={'horizontalFacet'} facet={facets?.find((facet) => facet.field === selectedFacet.field)!} />
```

Comment:
It appears that when using the `overlay` prop, that targeting a specific facet is not easily done. We may need additional classnames around each Dropdown component (selector should be similar to how the Facets / Facet component handle it.